plugin/src/main/java/io/jenkins/plugins/casc/yaml/YamlUtils.java

```diff
@@ -47,6 +53,23 @@ public static Node merge(List<YamlSource> configs) throws ConfiguratorException
         return root;
     }
 
+    /**
+     * Allows skipping yaml files we don't want to end up in the Jcasc config structure (e.g. secrets source in yaml file)
+     */
+    private static boolean isSourceAllowed(Node node) {
```

Comment:
I'm not a big fan of such special exclusions.
Would it make sense to get this new SecretSource configured by JCasC, just like we do in `SelfConfigurator`, before any other component get configured ?

Reply:
If I understand correctly you would create a SecretsConfigurator class that not really configures anything ?  The secrets do not have to be handled by the JCasC process itself.

Reply:
I mean a `YamlSecretSourceConfigurator` could consume this root `secrets` key to get configured early, then could provide secret to other configurators

Reply:
To get this clear:  `YamlSecretSourceConfigurator` would then feed it's data into the `ConfigurationContext` so e.g. `PrimitiveConfigurator` can also ask variable replacement from the `ConfigurationContext` ?  So you end up with 2 sources for variable replacement.  Seems not such a great design as well.